cns/restserver/ipam_test.go

```diff
@@ -214,35 +270,58 @@ func TestIPAMGetAvailableIPConfig(t *testing.T) {
 		t.Fatal("Expected IP retrieval to be nil")
 	}
 
-	desiredState := NewPodState(testIP1, 24, testPod1GUID, testNCID, types.Assigned, 0)
-	desiredState.PodInfo = testPod1Info
+	desiredState := make([]cns.IPConfigurationStatus, len(ncIDs))
+	for i := range ncIDs {
+		desiredState[i] = NewPodState(newPodIPs[i], prefixes[i], newPodIPs[i], ncIDs[i], types.Assigned, 0)
+		desiredState[i].PodInfo = testPod1Info
+	}
+
+	firstAddress, _ := netip.ParseAddr(actualstate[0].IPAddress)
```

Comment:
I'll add a comment. The reason I added this is because when we get IPs I can't consistently get it to always put the IPv4 one which is how the desired state is setup so I am checking if it is the case that IPv4 is first and if not then I switch them around.